tests/Feature/LoginTest.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Tests\Feature;
+
+use Carbon\Carbon;
+use Tests\TestCase;
+use App\Models\User;
+use App\Traits\InteractsWithJWT;
+use Illuminate\Foundation\Testing\TestResponse;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class LoginTest extends TestCase
+{
+    use RefreshDatabase, InteractsWithJWT;
+
+    /**
+     * @var User
+     */
+    protected $user;
+
+    /**
+     * Setup the test environment.
+     *
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->user = factory(User::class)->create([
+            'email' => 'test@example.org',
+            'password' => bcrypt('password'),
+        ]);
+    }
+
+    /**
+     * @param  array  $override
+     * @return array
+     */
+    private function validParamaters($override = [])
+    {
+        return array_merge([
+            'email' => 'test@example.org',
+            'password' => 'password',
+        ], $override);
+    }
+
+    /** @test */
+    public function a_user_can_login_and_retrieve_a_valid_token()
+    {
+        $request = $this->postJson(route('auth.login'), $this->validParamaters());
+
+        $request->assertStatus(200);
+
+        $token = $request->getData()->token;
+        $this->assertNotNull($token);
+
+        $decodedToken = $this->decodeJWT($token);
+        $this->assertEquals($this->user->getKey(), $decodedToken->sub);
+    }
+
+    /** @test */
+    public function an_email_is_required_to_login()
+    {
+        $request = $this->postJson(route('auth.login'), $this->validParamaters([
+            'email' => null,
+        ]));
+
+        $request->assertJsonValidationErrors('email');
```

Comment:
For this and all other places we're checking `assertJsonValidationErrors` can we also `assertStatus(422)`?

Also, for this and all other places we're assigning `$request` to `$this->HTTPMETHOD()` can we change that to `$response`?